IdentityCore/src/throttling/model/MSIDThrottlingModelInteractionRequire.m

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDThrottlingModelInteractionRequire.h"
+#import "MSIDThrottlingMetaData.h"
+#import "MSIDThrottlingMetaDataCache.h"
+#import "NSDate+MSIDExtensions.h"
+#import "NSError+MSIDExtensions.h"
+static NSInteger const DefaultUIRequired = 120;
+
+@implementation MSIDThrottlingModelInteractionRequire
+
+- (instancetype)initWithRequest:(id<MSIDThumbprintCalculatable>)request
+                    cacheRecord:(MSIDThrottlingCacheRecord *)cacheRecord
+                  errorResponse:(NSError *)errorResponse
+                    datasource:(id<MSIDExtendedTokenCacheDataSource> _Nonnull)datasource
+{
+    self = [super initWithRequest:request cacheRecord:cacheRecord errorResponse:errorResponse datasource:datasource];
+    if (self)
+    {
+        self.thumbprintType = MSIDThrottlingThumbprintTypeFull;
+        self.thumbprintValue = [request fullRequestThumbprint];
+        self.throttleDuration = DefaultUIRequired;
+    }
+    return self;
+}
+
+
+/**
+ * if is appliable for UIRequired:
+ * Throttle conditions:
+ * MSAL <-> server flow: OAuth error: error.msidOauthError is not nil
+ * MSAL <-> SSO-Ext <-> Server flow: error.code MSALErrorInteractionRequired
+ */
++ (BOOL)isApplicableForTheThrottleModel:(NSError *)errorResponse
+{
+    // MSALErrorInteractionRequired = -50002
+    NSSet *uirequiredErrors = [[NSSet alloc] initWithArray:@[[NSNumber numberWithInt:(-50002)]]];
```

Comment:
nit: instead of numberWithInt and numberWithLong, you can just use @() literal :)

Reply:
Thanks :) I Updated in the last commit